packages/compartment-mapper/src/powers.js

```diff
@@ -18,7 +18,8 @@ export const unpackReadPowers = powers => {
   if (typeof powers === 'function') {
     read = powers;
   } else {
-    ({ read, maybeRead, canonical } = powers);
+    ({ read, maybeRead, canonical } =
+      /** @type {import('./types.js').MaybeReadPowers} */ (powers));
```

Comment:
Cool.

Reply:
I think this is actually `MaybeReadPowers | ReadPowers`, but we don't know until L29